htdocs/projet/element.php

```diff
@@ -976,7 +988,11 @@
     				    $date = ($element->date_commande ? $element->date_commande : $element->date_valid);
     				}
     				elseif ($tablename == 'supplier_proposal') $date = $element->date_validation; // There is no other date for this
-    				elseif ($tablename == 'fichinter') $date = $element->datev; // There is no other date for this
+    				elseif ($tablename == 'fichinter')
+                                {
+                                    $dateo=$element->dateo;
```

Comment:
What is the dateo of an intervention ?

Reply:
Hello @eldy 

Based on what I read in other tables like the project ones dateo look like the start/begin of something.

``´
# https://wiki.dolibarr.org/index.php/Table_llx_projet
dateo | date | YES |   | NULL |   | date start project
```

I see it like the real start of the intervention, like you can create it begin of the week but the effective time is only from the first "line" until the validation.

Cheers

Reply:
A project has a start and an end date, sot there is field to store that.
But an intervention has no start and end date. It is the line of intervention that has a start and end date. That's the reason why there is no dateo for intervention.

Reply:
hello @eldy,

i work with @kadogo and we use extensively the "vue d'ensemble" in french and one of the big defect is that its not showing the start and end of intervention, but only the date of closure of the intervention who is not the most important info and doesnt help when trying to find an intervention started at certain date.

our patch take the creation date of the intervention and put that in a column to show the creation date on the list as you can see in the attached snapshot. 

its very usefull in daily work routine for the team

Cheers
![Screenshot_dolibarr-patch-intervention-vue-ensemble-date-intervention](https://user-images.githubusercontent.com/243766/73885620-65cdb480-4868-11ea-922d-b0e7b4e36916.png)
![Screenshot_dolibarr-patch-intervention-vue-ensemble](https://user-images.githubusercontent.com/243766/73885621-66664b00-4868-11ea-93ab-57e7153ebaf1.png)

Reply:
The "vue d'ensemble" does not show the start and end date of an intervention because such dates does not exists. Or said differently, there is several start date and several end dates.
1 intervention is a group of intervention line. Each intervention line has a start date and a duration, so and end date too. 
If you want to find an "intervention started at a given date", it means you want an intervention line and not an intervention group. It is same with contract and contract line.
If you add the field "date of intervention line", the the list of intervention will duplicate line for each intervention and the start date will be visible.

If you also need the details of each line on the "vue d'ensemble", then it means what you need is to show the intervention line so object FichinterLigne instead of Fichinter, so another object than intervention. Because on intervention the property ->dateo just does no exists.

Reply:
Hello @eldy 

> The "vue d'ensemble" does not show the start and end date of an intervention because such dates does not exists. Or said differently, there is several start date and several end dates.

Not based the code if I'm correct, dateo and datee are defined at https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fichinter/class/fichinter.class.php#L1649 and https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fichinter/class/fichinter.class.php#L1661

From my understanding I understand the next step of the code.

*  Grabbing all fichinterline for one fichinter and determine the elements
> * total_duration (sum of all duration of fichinterline)
> * dateo (minimum date of fichinterline)
> * datee (maxiumum date of fichinterline)

* Update the informations in fichinter
> * duree (equal to total_duration)
> * dateo (we use obj->dateo and obj is defined by our sql request before https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fichinter/class/fichinter.class.php#L1657)
> * datee (we use obj->datee and obj is still defined by our previous sql request)

Correct me if I'm wrong, but based on the code fichinter->dateo an fichinter->datee are updated based on the min / max date of the group of fichinterline.

> 1 intervention is a group of intervention line. Each intervention line has a start date and a duration, so and end date too.
> If you want to find an "intervention started at a given date", it means you want an intervention line and not an intervention group. It is same with contract and contract line.
> If you add the field "date of intervention line", the the list of intervention will duplicate line for each intervention and the start date will be visible.
> 
> If you also need the details of each line on the "vue d'ensemble", then it means what you need is to show the intervention line so object FichinterLigne instead of Fichinter, so another object than intervention. Because on intervention the property ->dateo just does no exists.

The values exit and are updated by the code if I'm correct, if not please correct me.

Reply:
> Hello @eldy
> 
> > The "vue d'ensemble" does not show the start and end date of an intervention because such dates does not exists. Or said differently, there is several start date and several end dates.
> 
> Not based the code if I'm correct, dateo and datee are defined at https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fichinter/class/fichinter.class.php#L1649 and https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fichinter/class/fichinter.class.php#L1661
> 

This is very old code, used when there was no line on intervention. Currently it is "dead" code because this request should insert always null into this 2 fields.


> From my understanding I understand the next step of the code.
> 
> * Grabbing all fichinterline for one fichinter and determine the elements
> 
> > * total_duration (sum of all duration of fichinterline)
> > * dateo (minimum date of fichinterline)
> > * datee (maxiumum date of fichinterline)
> 
> * Update the informations in fichinter
> 
> > * duree (equal to total_duration)
> > * dateo (we use obj->dateo and obj is defined by our sql request before https://github.com/Dolibarr/dolibarr/blob/develop/htdocs/fichinter/class/fichinter.class.php#L1657)
> > * datee (we use obj->datee and obj is still defined by our previous sql request)
> 
> Correct me if I'm wrong, but based on the code fichinter->dateo an fichinter->datee are updated based on the min / max date of the group of fichinterline.
> 
> > 1 intervention is a group of intervention line. Each intervention line has a start date and a duration, so and end date too.
> > If you want to find an "intervention started at a given date", it means you want an intervention line and not an intervention group. It is same with contract and contract line.
> > If you add the field "date of intervention line", the the list of intervention will duplicate line for each intervention and the start date will be visible.
> > If you also need the details of each line on the "vue d'ensemble", then it means what you need is to show the intervention line so object FichinterLigne instead of Fichinter, so another object than intervention. Because on intervention the property ->dateo just does no exists.
> 
> The values exit and are updated by the code if I'm correct, if not please correct me.

Reply:
We don't want to use the field dateo and datee. They are no more used and won't be used anymore. The information is already stored into fichinterdet and we should not duplicate information at different place. The start of an intervention is the min of the intervention lines.
The end, the max.

Reply:
Hello @eldy 

So based on your last sentence

> The start of an intervention is the min of the intervention lines.
> The end, the max.

It would be ok if the display use fichinterdet->min(...)  or fichinterdet->max(...) ?

Reply:
Yes, this is more clear like that.
It would be ok if we display use min_of_date_among_fichinterdet(...) or max_of_date_among_fichinterdet(...)